components/sup/src/manager/mod.rs

```diff
@@ -413,15 +461,107 @@ impl Manager {
                     }
                     Err(err) => warn!("Error loading service config after update, err={}", err),
                 }
-                self.state.butterfly.insert_service(rumor);
+                self.butterfly.insert_service(rumor);
                 last_update.service_counter += 1;
             }
         }
     }
 
+    fn persist_state(&self) {
+        debug!("Writing census state to disk");
+        self.persist_census_state();
+        debug!("Writing butterfly state to disk");
+        self.persist_butterfly_state();
+        debug!("Writing services state to disk");
+        self.persist_services_state();
+    }
+
+    fn persist_census_state(&self) {
+        let tmp_file = self.fs_cfg.census_data_path.with_extension("dat.tmp");
+        let file = match OpenOptions::new()
```

Comment:
Using `OpenOptions` is fine, but did you know that [`File::create`](https://doc.rust-lang.org/std/fs/struct.File.html#method.create) implements the same behavior (opens in write-only, creates file if not exists, otherwise truncates)?